api/src/main/java/org/phenotips/mendelianSearch/internal/DefaultPatientViewFactory.java

```diff
@@ -0,0 +1,236 @@
+            patient = this.pr.getPatientByExternalId(id);
+        }
+        if ((patient == null) || (variants == null) || (phenotypeScore == null)) {
+            return null;
+        }
+        boolean hasAccess = this.hasPatientAccess(patient);
+        if (hasAccess) {
+            view = this.generateOpenPatientView(patient, variants, phenotypeScore, request);
+        } else {
+            view = this.generateRestrictedPatientView(patient, variants, phenotypeScore, request);
+        }
+        return view;
+    }
+
+    @Override
+    public List<PatientView> createPatientViews(Set<String> ids, Map<String, List<GAVariant>> variantMap,
+        Map<String, Double> scores, MendelianSearchRequest request)
+    {
+        List<PatientView> result = new ArrayList<PatientView>();
+        if (ids == null || ids.isEmpty()) {
+            return result;
+        }
+        for (String id : ids) {
+            List<GAVariant> variants = variantMap.get(id);
+            Double phenotypeScore = scores.get(id);
+            PatientView view = this.createPatientView(id, variants, phenotypeScore, request);
+            if (view != null) {
+                result.add(view);
+            }
+        }
+        return result;
+    }
+
+    private PatientView generateRestrictedPatientView(Patient patient, List<GAVariant> variants,
+        double phenotypeScore, MendelianSearchRequest request)
+    {
+        PatientView view = new DefaultPatientView();
+        view.setType("restricted");
+        view.setPatientId(UNDISCLOSED_MARKER);
+        view.setOwner(UNDISCLOSED_MARKER);
+        view.setPhenotype(new ArrayList<String>());
+        view.setPhenotypeScore(phenotypeScore);
+        view.setVariants(new ArrayList<GAVariant>());
+        return view;
+    }
+
+    private PatientView generateOpenPatientView(Patient patient, List<GAVariant> variants,
+        double phenotypeScore, MendelianSearchRequest request)
+    {
+        PatientView view = new DefaultPatientView();
+        view.setType("open");
+
+        String id = patient.getId();
+        String owner = this.pm.getPatientAccess(patient).getOwner().getUsername();
+        List<String> phenotype = this.getDisplayedPatientPhenotype(patient);
+
+        view.setPatientId(id);
+        view.setPatientURL(this.getPatientURL(patient));
+        view.setGeneStatus(this.getPatientGeneStatus(patient, (String) request.get("geneSymbol")));
+        view.setOwner(owner);
+        view.setDiagnosis(this.getPatientDiagnosis(patient));
+
+        view.setPhenotype(phenotype);
+        view.setPhenotypeScore(phenotypeScore);
+        view.setVariants(variants);
+
+        return view;
+    }
+
+    private String getPatientURL(Patient patient)
+    {
+        XWiki xWiki = this.xcontext.get().getWiki();
+        return xWiki.getURL(patient.getDocument(), "view",
+            this.xcontext.get());
+    }
+
+    private List<String> getDisplayedPatientPhenotype(Patient patient)
```

Comment:
In general, if you dont have tests, you should at least have decent documentation for your internal (private) functions.